Content.Server/GameObjects/EntitySystems/Atmos/GasTileOverlaySystem.cs

```diff
@@ -0,0 +1,350 @@
+            {
+                data.Reset();
+            }
+        }
+
+        private void OnPlayerStatusChanged(object sender, SessionStatusEventArgs e)
+        {
+            if (e.NewStatus != SessionStatus.InGame)
+            {
+                if (_knownPlayerChunks.ContainsKey(e.Session))
+                {
+                    _knownPlayerChunks.Remove(e.Session);
+                }
+
+                return;
+            }
+
+            if (!_knownPlayerChunks.ContainsKey(e.Session))
+            {
+                _knownPlayerChunks[e.Session] = new PlayerGasOverlay();
+            }
+        }
+
+        /// <summary>
+        ///     Checks whether the overlay-relevant data for a gas tile has been updated.
+        /// </summary>
+        /// <param name="gam"></param>
+        /// <param name="oldTile"></param>
+        /// <param name="indices"></param>
+        /// <param name="overlayData"></param>
+        /// <returns>true if updated</returns>
+        private bool TryRefreshTile(GridAtmosphereComponent gam, GasOverlayData oldTile, MapIndices indices, out GasOverlayData overlayData)
+        {
+            var tile = gam.GetTile(indices);
+            var tileData = new List<GasData>();
+
+            for (var i = 0; i < Atmospherics.TotalNumberOfGases; i++)
+            {
+                var gas = Atmospherics.GetGas(i);
+                var overlay = Atmospherics.GetOverlay(i);
+                if (overlay == null || tile.Air == null) continue;
+
+                var moles = tile.Air.Gases[i];
+
+                if (moles == 0.0f || moles < gas.GasMolesVisible) continue;
+
+                var data = new GasData(i, MathF.Max(MathF.Min(1, moles / gas.GasMolesVisibleMax), 0f));
+                tileData.Add(data);
+            }
+
+            overlayData = new GasOverlayData(tile.Hotspot.State, tile.Hotspot.Temperature, tileData.Count == 0 ? null : tileData.ToArray());
+
+            if (overlayData.Equals(oldTile))
+            {
+                return false;
+            }
+            
+            return true;
+        }
+
+        /// <summary>
+        ///     Get every chunk in range of our entity that exists, including on other grids.
+        /// </summary>
+        /// <param name="entity"></param>
+        /// <returns></returns>
+        private IEnumerable<GasOverlayChunk> GetChunksInRange(IEntity entity)
+        {
+            var entityTile = _mapManager
+                .GetGrid(entity.Transform.GridID)
+                .GetTileRef(entity.Transform.GridPosition)
+                .GridIndices;
+
+            // This is the max in any direction that we can get a chunk (e.g. max 2 chunks away of data).
+            var (maxXDiff, maxYDiff) = ((int) (UpdateRange / ChunkSize) + 1, (int) (UpdateRange / ChunkSize) + 1);
+            
+            var worldBounds = Box2.CenteredAround(entity.Transform.WorldPosition,
+                new Vector2(20.0f, 20.0f));
```

Comment:
What is this 20, 20? Is it a hardcoded max view size?

Reply:
That can actually be changed. Originally I had the client deleting their own chunks as required and I had hardcoded it until server -> client cvars are in, though now the client retains chunks indefinitely so they don't need the cvar.

Should I make a separate cvar for atmos updates or use the existing net.maxupdaterange? (Bearing in mind due to the fact the overlay update isn't run every tick will need to be higher than the PVS range).

Reply:
Most thorough would be basing it off `net.maxupdaterange` and then having a bias (either constant or cvar) on top of that, probably.